static/js/components/AddToListDialog.js

```diff
@@ -100,8 +100,8 @@ export default function AddToListDialog() {
     const matchingList = find(propEq("list_id", userListId), resource.lists)
     if (checked && !matchingList) {
       resource.lists.push({ list_id: userListId })
-    } else {
-      resource.lists.pop(matchingList)
+    } else if (matchingList && !checked) {
+      resource.lists.splice(resource.lists.indexOf(matchingList), 1)
     }
   }
```

Comment:
is this `updateResource` basically just to modify the object on the frontend in order to render the right UI?

Reply:
just wondering if there might be a better way to do this overall - i.e. if we're going to be editing the object it is probably better to that in the redux store itself rather than indirectly here

Reply:
Yes (1st comment), maybe? (2nd) - not sure what the best way of doing that is, the userlist entitities are being modified/returned in the API calls rather than the resource being added to those lists.  If I modify `updateResource` to be something like:
```
  const [, updateResource] = useRequest(
    resource ? courseRequest(resource.id) : null
  )
```
then the dialog is closed as soon as I check or uncheck a box.

Reply:
yeah that would fetch a new one, I don't think we need to do that but if we wanted to we could try to do a modification in the `update` method for the `userListMutation`. it's just probably not the best thing to modify the object here, but there's not an easy way to do that so I think it's probably ok